src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/api/impl/Channel.kt

```diff
@@ -168,11 +170,13 @@ class Channel(
             // Date from buffer should be copied for prost-processing (mangler.postOutgoing and onMessage handling).
             // The post-processing is executed asynchronously after sending message via tcp channel where original buffer is released
             val data = Unpooled.copiedBuffer(buffer).asReadOnly()
+            val sendingTimestamp = Instant.now()
```

Comment:
this is incorrect place for taking operation timestamp, because TcpChannel.send method includes the code for waiting active  io.netty.channel.Channel
`while (!channel.isWritable && channel.isActive && --timeout != 0L) Thread.sleep(1)`
Could we take timestamp in the TcpChannel.send method on the line before channel.writeAndFlush(data)

Reply:
I don't think we can capture it there. However, we capture time in the ChannelFuture callback and pass it to the lambda that process the buffer sent